From dbcdde3e76f45d56c4a30ca6c5d69b6d473d3fd1 Mon Sep 17 00:00:00 2001
From: Christoph Hellwig <hch@lst.de>
Date: Fri, 8 Jul 2011 14:34:14 +0200
Subject: [PATCH] xfs: re-enable non-blocking behaviour in xfs_map_blocks
Git-commit: dbcdde3e76f45d56c4a30ca6c5d69b6d473d3fd1
Patch-mainline: 3.1-rc5
References: bnc#900279

mhocko@suse.cz:
This has been tested as a part of bnc#900279 but it hasn't shown any big
improvements. It is a good to have fix anyway so let's add it as well.
The credit for the backp goes to Neil Brown.

The non-blockig behaviour in xfs_vm_writepage currently is conditional on
having both the WB_SYNC_NONE sync_mode and the nonblocking flag set.
The latter used to be used by both pdflush, kswapd and a few other places
in older kernels, but has been fading out starting with the introduction
of the per-bdi flusher threads.

Enable the non-blocking behaviour for all WB_SYNC_NONE calls to get back
the behaviour we want.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Alex Elder <aelder@sgi.com>
Reviewed-by: Dave Chinner <dchinner@redhat.com>
Acked-by: Jan Kara <jack@suse.cz>

---
 fs/xfs/linux-2.6/xfs_aops.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.0-SLE11-SP3.orig/fs/xfs/linux-2.6/xfs_aops.c
+++ linux-3.0-SLE11-SP3/fs/xfs/linux-2.6/xfs_aops.c
@@ -976,7 +976,7 @@ xfs_vm_writepage(
 	offset = page_offset(page);
 	type = IO_OVERWRITE;
 
-	if (wbc->sync_mode == WB_SYNC_NONE && wbc->nonblocking)
+	if (wbc->sync_mode == WB_SYNC_NONE)
 		nonblocking = 1;
 
 	do {
